src/bun.js/bindings/JSBuffer.cpp

```diff
@@ -995,20 +1004,139 @@ static inline JSC::EncodedJSValue jsBufferPrototypeFunction_fillBody(JSC::JSGlob
     }
 }
 
-static inline JSC::EncodedJSValue jsBufferPrototypeFunction_includesBody(JSC::JSGlobalObject* lexicalGlobalObject, JSC::CallFrame* callFrame, typename IDLOperation<JSBuffer>::ClassParameter castedThis)
+static int indexOf(const uint8_t* thisPtr, size_t thisLength, const uint8_t* valuePtr, size_t valueLength, int byteOffset)
+{
+    if (thisLength < valueLength + byteOffset)
+        return -1;
+    auto start = thisPtr + byteOffset;
+    auto end = thisPtr + thisLength;
+    auto it = static_cast<uint8_t*>(memmem(start, thisLength - byteOffset, valuePtr, valueLength));
+    if (it != NULL) {
+        return it - thisPtr;
+    }
+    return -1;
+}
+
+static int lastIndexOf(const uint8_t* thisPtr, size_t thisLength, const uint8_t* valuePtr, size_t valueLength, int byteOffset)
+{
+    auto start = thisPtr;
+    auto end = thisPtr + std::min(thisLength, byteOffset + valueLength);
+    auto it = std::find_end(start, end, valuePtr, valuePtr + valueLength);
+    if (it != end) {
+        return it - thisPtr;
+    }
+    return -1;
+}
+
+static int indexOf(JSC::JSGlobalObject* lexicalGlobalObject, JSC::CallFrame* callFrame, typename IDLOperation<JSBuffer>::ClassParameter castedThis, bool last)
```

Comment:
typed arrays go up to 4 GB, which a signed integer will overflow

Instead, lets use `int64_t`

Reply:
sure~ I've changed them all to `int64_t`, including those `ssize_t` comments below and only cast back to `size_t` in `memmem`. In this way, we could prevent the unexpected conversions.